src/ontology/mondo-ingest.Makefile

```diff
@@ -160,15 +160,34 @@ mappings: sssom $(ALL_MAPPINGS)
 #################
 # Utils #########
 #################
-# Documentation for this commands in this section is in: `docs/developer/ordo.md`
+# Documentation for `report-mapping-annotations` and `update-jinja-sparql-queries`: `docs/developer/ordo.md`
+# TODO: When https://github.com/monarch-initiative/mondo-ingest/issues/43 is fixed, can change back to `requirements.txt`
+python-install-dependencies:
+	python3 -m pip install --upgrade pip
+	python3 -m pip install -r $(RELEASEDIR)/requirements-unlocked.txt
 
-report-mapping-annotations:
+report-mapping-annotations: python-install-dependencies
 	python3 $(SCRIPTSDIR)/ordo_mapping_annotations/report_mapping_annotations.py
 
-update-jinja-sparql-queries:
+update-jinja-sparql-queries: python-install-dependencies
 	python3 $(SCRIPTSDIR)/ordo_mapping_annotations/create_sparql__ordo_replace_annotation_based_mappings.py
 	python3 $(SCRIPTSDIR)/ordo_mapping_annotations/create_sparql__ordo_mapping_annotations_violation.py
 
+# TODO: Need to add dependency for sig/mirror files?
+
+# TODO: update my script to accept this positional param $@
+# note: why not use $@: will only pick the calling target (e.g. if someone calls for the first tone)
+# - try this
+
+# TODO: temp for when I don't want `python-install-dependencies`
+# $(REPORTDIR)/excluded_terms_in_mondo_xrefs.tsv $(REPORTDIR)/excluded_terms_in_mondo.tsv: $(TMPDIR)/mondo.sssom.tsv python-install-dependencies
+$(REPORTDIR)/excluded_terms_in_mondo_xrefs.tsv $(REPORTDIR)/excluded_terms_in_mondo.tsv: $(TMPDIR)/mondo.sssom.tsv
```

Comment:
@matentzn I have been playing with having multiple targets as the make goal name, e.g. in this case `$(REPORTDIR)/excluded_terms_in_mondo_xrefs.tsv $(REPORTDIR)/excluded_terms_in_mondo.tsv:`.

## Summary
Bad design probably because file names could change.

## Discussion
I will keep this design unless you want it changed, but I think it's probably not the best design in this case.

I'm just starting to figure out how this works, and I ran an experiment:
```
make reports/excluded_terms_in_mondo_xrefs.csv reports/excluded_terms_in_mondo.csv

python3 ../analysis/excluded_terms_in_mondo.py \
	--mondo-mappings-path tmp/mondo.sssom.tsv \
	--outpath reports/excluded_terms_in_mondo_xrefs.tsv
Testing positional args passed in from makefile: reports/excluded_terms_in_mondo_xrefs.tsv

python3 ../analysis/excluded_terms_in_mondo.py \
	--mondo-mappings-path tmp/mondo.sssom.tsv \
	--outpath reports/excluded_terms_in_mondo.tsv
Testing positional args passed in from makefile: reports/excluded_terms_in_mondo.tsv
```

Basically my problem is that my script needs to understand the filename being passed to `--outpath`. I think that's bad design because the filename could change.

## Possible solutions
Given that, in this case, probably the best thing to do is to have (a) 2 different Python scripts and 2 different make goals, or (b) 1 python script with 2 different make goals, and each goal calls different args: `--outpath-excluded-in-mondo` for one, and `--outpath-excluded-xrefed-in-mondo` for the other.

I prefer (b). It's probably best to keep it all in one script since the logic for both jobs is very related.

If you like this proposal, let me know and I will do it.

Reply:
Nico decided that we only need to report on excluded classes that are still xref'ed in mondo for this work, so this is no longer an issue.